near-sdk-macros/src/lib.rs

```diff
@@ -786,7 +775,9 @@ pub fn derive_no_default(item: TokenStream) -> TokenStream {
 
 /// `BorshStorageKey` generates implementation for `BorshIntoStorageKey` trait.
 /// It allows the type to be passed as a unique prefix for persistent collections.
-/// The type should also implement or derive `BorshSerialize` trait.
+/// The type should also implement or derive `BorshSerialize` trait. 
+/// 
+/// More information about storage keys in [NEAR documentation](https://docs.near.org/build/smart-contracts/anatomy/storage)
```

Comment:
the same as above, the documentation page may be removed or changed someday, let's add a complete example

```rs
#[derive(BorshSerialize, BorshDeserialize, BorshStorageKey)]
#[borsh(crate = "near_sdk::borsh")]
pub enum StorageKey {
    Messages,
}

// Define the contract structure
#[near(contract_state)]
pub struct Contract {
    messages: Vector<String>
}

// Define the default, which automatically initializes the contract
impl Default for Contract {
    fn default() -> Self {
        Self {
            messages: Vector::new(StorageKey::Messages)
        }
    }
}
```

Reply:
Thanks for the detailed examples!